docs/src/tutorials/getting_started/transitioning_from_matlab.jl

```diff
@@ -0,0 +1,280 @@
+# arrays are rather slow. This is not a problem in Julia: a vector of matrices
+# is almost as fast as a 3-dimensional array.
+
+# ## Declaring constraints
+
+# As in the case of variables, in most cases there is a direct translation
+# between the syntaxes:
+
+# |                     | JuMP                                                     | YALMIP               | CVX                                      |
+# | ------------------- | -------------------------------------------------------- | -------------------- | ---------------------------------------- |
+# | equality constraint | `@constraint(model, v == c)`                             | `v == c`             | `v == c`                                 |
+# | nonnegative cone    | `@constraint(model, v >= 0)`                             | `v >= 0`             | `v >= 0`                                 |
+# | real PSD cone       | `@constraint(model, m in PSDCone())`                     | `m >= 0`             | `m == semidefinite(length(m))`           |
+# | complex PSD cone    | `@constraint(model, m in HermitianPSDCone())`            | `m >= 0`             | `m == hermitian_semidefinite(length(m))` |
+# | second order cone   | `@constraint(model, [t; v] in SecondOrderCone())`        | `cone(v, t)`         | `{v, t} == lorentz(length(v))`           |
+# | exponential cone    | `@constraint(model, [x, y, z] in MOI.ExponentialCone())` | `expcone([x, y, z])` | `{x, y, z} == exponential(1)`            |
+
+# A subtlety appears when declaring equality constraints for matrices. In
+# general, JuMP uses `@constraint(model, m .== c)`, with the dot meaning
+# broadcasting in Julia, except when `m` is `Symmetric` or `Hermitian`: in this
+# case `@constraint(model, m == c)` is allowed, and is much better, as JuMP is
+# smart enough to not generate redundant constraints for the lower diagonal and
+# the imaginary part of the diagonal (in the complex case). Both YALMIP and CVX
+# are also smart enough to do this and the syntax is always just `m == c`.
+
+# Experienced YALMIP users will probably be relieved to see that `>=` is only
+# ever used to make a vector nonnegative, never to make a matrix positive
+# semidefinite, as this ambiguity is reliable source of bugs.
+
+# Like CVX, but unlike YALMIP, JuMP can also constrain variables upon creation:
+
+# |                    | JuMP                                                 | CVX                                    |
+# | ------------------ | ---------------------------------------------------- | -------------------------------------- |
+# | nonnegative vector | `@variable(model, v[1:d] >= 0)`                      | `variable v(d) nonnegative`            |
+# | real PSD matrix    | `@variable(model, m[1:d,1:d] in PSDCone())`          | `variable m(d,d) semidefinite`         |
+# | complex PSD matrix | `@variable(model, m[1:d,1:d] in HermitianPSDCone())` | `variable m(d,d) complex semidefinite` |
```

Comment:
What about mentioning variable bounds, integrality, starting points, etc?

Reply:
I don't know what you mean. Variable bound? Like `x <= 1`? Sure I could mention it explicitly. Integrality is what, constraining variables to be bits or integers? That's a single line to add in the table. Starting point is setting the starting point of the optimizer? I wouldn't do that, it hardly ever works.

Reply:
Some extra details could go into an session by the end of the tutorial to do break the flow.

Reply:
I added rows for binary and integer, and a section for bounds. I decided not to add starting points.